cmd/admin-netperf.go

```diff
@@ -0,0 +1,163 @@
+/*
+ * MinIO Client (C) 2019 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"fmt"
+	"net/url"
+	"sort"
+
+	"github.com/fatih/color"
+	"github.com/minio/cli"
+	json "github.com/minio/mc/pkg/colorjson"
+	"github.com/minio/mc/pkg/console"
+	"github.com/minio/mc/pkg/probe"
+	"github.com/minio/minio/pkg/madmin"
+)
+
+var (
+	adminNetPerfFlags = []cli.Flag{}
+)
+
+var adminNetPerfCmd = cli.Command{
+	Name:            "netperf",
+	Usage:           "network performance of all cluster nodes",
+	Action:          mainAdminNetPerf,
+	Before:          setGlobalsFromContext,
+	Flags:           append(adminNetPerfFlags, globalFlags...),
+	HideHelpCommand: true,
+	CustomHelpTemplate: `Name:
+  {{.HelpName}} - {{.Usage}}
+
+USAGE:
+  {{.HelpName}} TARGET
+
+FLAGS:
+  {{range .VisibleFlags}}{{.}}
+  {{end}}
+EXAMPLES:
+  1. Get network performance of all cluster nodes by 'myminio' server.
+     $ {{.HelpName}} myminio/
+`,
+}
+
+// netPerfMessage holds network performance information.
+type netPerfMessage struct {
+	Addr    string               `json:"address"`
+	Err     string               `json:"error"`
+	NetPerf []madmin.NetPerfInfo `json:"netPerf"`
+}
+
+func (s netPerfMessage) JSON() string {
+	if len(s.NetPerf) == 0 {
+		s.Err = "cannot fetch network performance stats"
+	}
+
+	statusJSONBytes, e := json.MarshalIndent(s, "", " ")
+	fatalIf(probe.NewError(e), "Unable to marshal into JSON")
+
+	return string(statusJSONBytes)
+}
+
+func (s netPerfMessage) String() string {
+	if len(s.NetPerf) == 0 {
+		msg := fmt.Sprintf("%s  %s\n", console.Colorize(monitorDegraded, "Source:"), s.Addr) +
+			fmt.Sprintf("   Error: %s\n", console.Colorize(monitorFail, "cannot fetch network performance stats"))
+		return msg
+	}
+
+	msg := fmt.Sprintf("%s  %s\n\n", console.Colorize(monitor, "Source:"), s.Addr)
```

Comment:
Can we remove this extra `\n`? 

![image](https://user-images.githubusercontent.com/5410427/64924032-c1d26680-d7fd-11e9-9e5d-af43ee1f6ce0.png)